src/react-components/src/helpers/hoc/WithOneChild.tsx

```diff
@@ -0,0 +1,55 @@
+import * as React from "react";
+
+/**
+ * Wrapper component to verify that a component only has one child.
+ * @param Component Component to test, either a React element
+ *   or an object element.
+ */
+const withOneChild = <P extends object>(
+  Component: React.ComponentType<P> | any
+): React.ComponentType<P> => (
+  ({ ...initProps }) => {
+    // It's more likely that the component passed will be a
+    // React element so set these as default.
+    let componentChildren = initProps.children;
+    let componentToRender = Component;
+    let name = Component.name;
+    let props = initProps;
+
+    // In case the component is an object!
+    // E.g, a component passed as a prop.
```

Comment:
I'm not completely sure what's happening here -- does this mean that this will work when a component is passed as a prop, like this:  https://github.com/NYPL/nypl-design-system/blob/development/src/react-components/src/components/02-molecules/Cards/EditionCard.tsx#L25 ?  

I'm wondering if we need this - every time a component needs to be passed as a prop it should be wrapped in something else (where the passed component is children).  Perhaps it would be clearer or simpler if we don't allow a component to be passed as a prop.

Reply:
That's right. It was to cover a similar case so it can work: https://github.com/NYPL/nypl-design-system/blob/HOC-error-children/src/react-components/src/components/02-molecules/Cards/SearchResultItem.tsx#L36

You're right, every time a component is passed it does need to be wrapped. It's could be a superfluous check, but I think covering that case _just in case_ is okay to do.